sled-agent/src/instance.rs

```diff
@@ -610,6 +615,37 @@ impl Instance {
             zone_bundler,
         } = services;
 
+        let mut dhcp_config = DhcpCfg {
+            hostname: Some(
+                hardware
+                    .properties
+                    .hostname
```

Comment:
Not necessarily for this PR, but I think it might be worth an filing issue: I'm guessing that if this conversion fails it implies that the hostname in the instance create parameters was completely bogus such that this instance will never be able to start. If that's so we should see about rejecting invalid hostnames in `Nexus::project_instance_create` so that we reject bad hostnames up front instead of letting the create go through.

Reply:
I haven't gone and verified this is the case, but I _believe_ that our restrictions on the `Name` type make them all valid single-label hostnames. It would be good to prove this out though (or use a single consistent type across Omicron and OPTE).

Reply:
Oh! Well. I'll see if that's not too bad to fix along with this PR.